ui/admin/app/components/form/policy/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+const RETENTION_POLICY = {
+  forever: -1,
+  custom: 1,
+  do_not_protect: 0,
+};
+
+const DELETION_POLICY = { do_not_delete: 0, custom: 1 };
+
+export default class FormPolicyComponent extends Component {
+  //attributes
+
+  @tracked retainForOverridable = this.args.model.retain_for?.overridable;
+  @tracked deleteAfterOverridable = this.args.model.delete_after?.overridable;
+  @tracked isCustomRetentionSelected = this.args.model.retain_for?.days > 0;
+  @tracked isCustomDeletionSelected = this.args.model.delete_after?.days > 0;
```

Comment:
Are these still needed if you can now directly use the model?

Reply:
We would still need to keep it there because I'm using these fields to display the custom inputs component for both `delete after` and `retain for`. If I were to move these to the policy-selection component, I won't be able to toggle and use a generic input to show/hide custom inputs.

Reply:
Hmm fair but don't you already do a check inside `handlePolicyTypeSelection` for the policy type based on name? What's the difference then of just checking which one you need based on name again?

Reply:
💯 yes, that is possible; I also moved the toggle checks along with these. we would only have some methods on the main form, and then the generic one will have the actions.